### DISCUSSION TITLE
How to check new data and show a notification

### DISCUSSION THREAD

**Discussion by kungfu321** [vercel/swr#1120] [Q&A]
I'm using SWR to fetch new data, but I want to show a notification when has new data was fetched.
Any idea for me. Thanks

**Comment by promer94** [comment]
I think you could use the useEffect hook to achieve this
```js
const { data } = useSWR('/api')
useEffect(() => {
 if(data) {
   // show notification
 } 
}, [data])
```

**Reply by kungfu321** [reply]
It does not work true. Because when the first data or change router, it shows a notification.

**Reply by shuding** [reply]
You can use a ref to keep the previous data, and compare it with the latest value.

**Comment by andykao1213** [comment]
Try to do something with the options `onSuccess`? `onSuccess` will only be fired if there is a request being made.
```jsx
useSWR('/api', { onSuccess: (data, key, config) => { /* do whatever you want */ }})
```

**Reply by kungfu321** [reply]
But how to know when the time have new data?
Because `onSuccess ` return all data

**Reply by nandorojo** [reply]
Can you compare it somehow?

```ts
const oldData = useRef()

useSWR('/api', { onSuccess: (data, key, config) => { 
  const didDataChange = data !== oldData.current // you can customize this
  if (didDataChange) {
    // do some thing here
  }
  oldData.current = data
}})
```

**Comment by kungfu321** [comment]
I found `compare(a, b)` options can compare when the data change, and it's work for me.

**Reply by VladSez** [reply]
@kungfu321 could you share your solution?

**Comment by mmortelliti** [comment]
You can probably also use something like `data.status !== 304...` since the response status of 304 would mean 'not modified'

**Comment by tettoffensive** [comment]
I'm looking for a way when the data has changed to trigger another call to refetch the data. I'm not sure this is a very good practice to do this way but...

```ts
{
      onSuccess: async (newData) => {
        const equal = dequal(oldData.current, newData);
        oldData.current = data;
        if (!equal) {
           await mutate(keyForADifferentMutation);
        }
      },
    },
```

I didn't use compare because I didn't want the `compare` method to wait for a Promise, but I was thinking it would be ok in `onSuccess`

**Reply by nandorojo** [reply]
There’s no need for this to be a promise. You aren’t calling anything after mutate, so you don’t have to await it. In any case, it’s fine for that to be a promise.
